micrometer-core/src/main/java/io/micrometer/core/instrument/binder/httpcomponents/DefaultApacheHttpClientObservationConvention.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.httpcomponents;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.common.lang.Nullable;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpResponse;
+
+/**
+ * Default implementation of {@link ApacheHttpClientObservationConvention}.
+ *
+ * @since 1.10.0
+ * @see ApacheHttpClientDocumentedObservation
+ */
+public class DefaultApacheHttpClientObservationConvention implements ApacheHttpClientObservationConvention {
+
+    /**
+     * Singleton instance of this convention.
+     */
+    public static final DefaultApacheHttpClientObservationConvention INSTANCE = new DefaultApacheHttpClientObservationConvention();
+
+    // There is no need to instantiate this class multiple times, but it may be extended,
+    // hence protected visibility.
+    protected DefaultApacheHttpClientObservationConvention() {
+    }
+
+    @Override
+    public String getName() {
+        return MicrometerHttpRequestExecutor.METER_NAME;
+    }
+
+    @Override
+    public String getContextualName(ApacheHttpClientContext context) {
+        return "HTTP " + getMethodString(context.getCarrier());
```

Comment:
I've seen this hard coded quite a few times - I wonder if we can extract this somehow, somewhere? :shrug:

Reply:
Since how to get the method will be different for each implementation I'm not sure we really can. This is exactly what we were trying to do with the HTTP abstractions for request/response, but we ended up going a different direction.